pkg/tiles/side/side.go

```diff
@@ -140,3 +140,53 @@ func (side Side) Rotate(rotations uint) Side {
 	var shift = rotations * 2
 	return (side >> shift) | (side << (8 - shift)) // circular bitshift (bitwise rotate) side to the right by {2*rotations} bits
 }
+
+/*
+argument indicates only ONE cardinal or edge, otherwise it's ambigous
+*/
+func (side Side) ConnectedOpposite() Side {
+
+	SwapBits := func(number Side, i uint8, j uint8) Side {
+		// check if needs to swap (check xor)
+		if (((number & (1 << i)) >> i) ^ ((number & (1 << j)) >> j)) == 1 {
+			// xor to swap
+			number ^= 1 << i
+			number ^= 1 << j
+		}
+		return number
+	}
+
+	// rotate
+	rotated := side.Rotate(2)
+
+	// mirror edges
+	for i := range 4 {
+		rotated = SwapBits(rotated, uint8(2*i), uint8(2*i+1))
+	}
+	return rotated
+}
+
+func (side Side) GetNthCardinalDirection(n uint8) Side {
+	cardinals := []Side{Top, Left, Right, Bottom}
```

Comment:
Please add a comment explaining what this function does.
Also, does it check the directions in a specific order, and does the order matters for anything? Why?

Reply:
commented, checking is in specific order, but reordering it wouldn't change anything. (iterating on all sides simply always in same order)

Reply:
Ok, I understand, but am still not sure about the order. For example, if `side = Left | Bottom`, side.GetNthCardinalDirection(1) would return Bottom, but with a different iteration order it could return Top.

I think what you really need instead of this function is something like "side.GetConnectedCardinalDirection(direction)" (or GetOtherCardinalDirection, something like that), which in this example would behave like so:
side.GetConnectedCardinalDirection(Left) = Bottom
side.GetConnectedCardinalDirection(Bottom) = Left

This could also allow rewriting this:
```
if road.Sides.GetNthCardinalDirection(0) == roadSide {
	roadSide = road.Sides.GetNthCardinalDirection(1)
} else {
	roadSide = road.Sides.GetNthCardinalDirection(0)
}
```
to a much simpler:
```
roadSide = road.Sides.GetConnectedCardinalDirection(roadSide)
```

Reply:
I left the function GetNthCardinalDirection, as it simplifies the beggining of the ScoreRoadCompletion function. I created the new function GetConnectedOtherCardinalDirection, which you described.